src/govuk/components/button/_index.scss

```diff
@@ -1,5 +1,16 @@
+////
+/// @group components/button
+////
+
+/// Button component primary background colour
+///
+/// @type Colour
+/// @access public
+
+$govuk-button-background-colour: govuk-colour("green", $legacy: #00823b) !default;
+
 @include govuk-exports("govuk/component/button") {
-  $govuk-button-colour: govuk-colour("green", $legacy: #00823b);
+  $govuk-button-colour: $govuk-button-background-colour;
   $govuk-button-hover-colour: govuk-shade($govuk-button-colour, 20%);
   $govuk-button-shadow-colour: govuk-shade($govuk-button-colour, 60%);
   $govuk-button-text-colour: govuk-colour("white");
```

Comment:
As we're allowing customisation of the background-color, it feels like we should also provide a mechanism to customise the text color so that it's possible to maintain accessible contrast.

Reply:
Agreed, this'll also be useful when either I or the GOV.UK Design System team writes about branding. My plan is to contribute these improvements based on real world usage of branding while doing client work. Then writing guidance on how this is done that could be contributed back to the main GOV.UK Design System if it becomes useful to do so.